src/cloud-providers/libvirt/provider.go

```diff
@@ -79,6 +79,11 @@ func (p *libvirtProvider) CreateInstance(ctx context.Context, podName, sandboxID
 	}
 	logger.Printf("LaunchSecurityType: %s", vm.launchSecurityType.String())
 
+	if spec.Image != "" {
+		logger.Printf("Choosing %s as libvirt volume for the PodVM image", spec.Image)
+		p.libvirtClient.volName = spec.Image
```

Comment:
Hi Ajay, I'd like to understand better how this change is propagated through if needed. My (very limited) understanding is that we set this value initially when we create the libvirt provider client, so it overriding that here sufficient to pick up this custom specified image, or is there any other code we need to skip to stop the previous version being used?

Reply:
Hi Steve,
Exactly, that's what my understanding also is..

With the enhancement of `CreateVM()` function to support getting the image from `GetImageFromAnnotation()`, we are able to retrieve the same on `CreateInstance()` and then furthermore override the default `Volume Name` which is set initially once the CAA comes up for the libvirt provider as you pointed. 

I guess with the `CreateDomain()` flow starting post overriding the volume and initially the volume being picked up from the `LIBVIRT_VOL_NAME` which is being assigned to `libvirtcfg.VolName`, we might not have anywhere else to update?

Kindly suggest as I might be missing something here.

Reply:
Thanks, there isn't anything I know that is missing, I just wanted to check my understanding.

I'm assuming that you've tested this locally. Is there a way to integrate this into our e2e tests for libvirt?